### PR TITLE
Moving some blockdevices examples from teams into this repo

### PR DESCRIPTION
**Pull request by yossi2le**
The MBRblockDevice, SlicingBlockDevice and ProfilingBlockDevice examples where out of date, not synced with the MBRBlockDevice.h Doxygen examples. therefore I have removed example from the Doxygen and moved the examples themselves into this repo.

### CODE REVIEW

File: MBRBlockDevice_ex_2/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# MBRBlockDevice example 2
+
+Partition an SD card, and format the new partition with a FAT filesystem. A PC will now be able to recognize the SD card.
```
Comment (melwee01) [inline MBRBlockDevice_ex_2/README.md]: The SD card is now recognizable to devices.

File: MBRBlockDevice_ex_2/main.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    // Format the partition with a FAT filesystem
+    printf("Format the partition with a FAT filesystem\n");
+    FATFileSystem::format(&part1);
+
+    // Create the FAT filesystem instance, files can now be written to
+    // the FAT filesystem in partition 1
+    printf("Create the FAT filesystem instance, files can now be written to the FAT filesystem in partition 1\n");
+    FATFileSystem fat("fat", &part1);
+
+    printf("You should be abale to read the SD on windows machine and verify if the exampe is working\n");
```
Comment (melwee01) [inline MBRBlockDevice_ex_2/main.cpp]: To verify that the example worked, read the SD on a Windows machine.

File: SlicingBlockDevice/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# SlicingBlockDevice example
+
+This SlicingBlockDevice example partitions a HeapBlockDevice into three subunits and showcases programming and reading back data segments through both the underlying master block device and the sliced subunits.
```
Comment (melwee01) [inline SlicingBlockDevice/README.md]:
```suggestion
This SlicingBlockDevice example partitions a HeapBlockDevice into three subunits. It showcases programming and reading back data segments, through both the underlying master block device and sliced subunits.
```

**Comment by cmonr** [comment]
@yossi2le Would it be possible to consolidate the MBRBlockDevice examples (or even all of the Block Device examples)?

You can put them into sub-directories, similar to this structure: https://github.com/ARMmbed/mbed-os-examples-docs_only/tree/master/Flow-Control

**Comment by cmonr** [comment]
Ty @yossi2le!

**Pull request merged**